### PR TITLE
Split `classpath` key into `classpath` and `finalClasspath`.

### PR DESCRIPTION
**Pull request by kjanosz**

### CODE REVIEW

File: scalameter-core/src/main/scala/org/scalameter/Context.scala [hunk 1]
```diff
@@ -20,6 +20,9 @@ case class Context(properties: immutable.Map[Key[_], Any]) {
     }
   }
   def goe[T](key: Key[T], v: => T) = properties.getOrElse(key, v).asInstanceOf[T]
+  def apply[T](key: KeyWithDefaultKey[T]) = {
+    properties.get(key).orElse(properties.get(key.default)).asInstanceOf[Option[T]].getOrElse(key.default.defaultValue)
```
Comment (axel22) [inline scalameter-core/src/main/scala/org/scalameter/Context.scala]: Please break this into several lines, respecting the 88 char limit.

File: scalameter-core/src/main/scala/org/scalameter/keys.scala [hunk 2]
```diff
@@ -39,7 +39,10 @@ class Key[T: Pickler](val name: String)(implicit container: KeyContainer) extend
 
 
 class KeyWithDefault[T: Pickler](name: String, val defaultValue: T)(implicit container: KeyContainer)
-extends Key[T](name)
+  extends Key[T](name)
+
+class KeyWithDefaultKey[T: Pickler](name: String, val default: KeyWithDefault[T])(implicit container: KeyContainer)
```
Comment (axel22) [inline scalameter-core/src/main/scala/org/scalameter/keys.scala]: Please add a one-line ScalaDoc comment for both these `Key` subclasses.

File: scalameter-core/src/main/scala/org/scalameter/Context.scala [hunk 3]
```diff
@@ -20,6 +20,9 @@ case class Context(properties: immutable.Map[Key[_], Any]) {
     }
   }
   def goe[T](key: Key[T], v: => T) = properties.getOrElse(key, v).asInstanceOf[T]
+  def apply[T](key: KeyWithDefaultKey[T]) = {
```
Comment (axel22) [inline scalameter-core/src/main/scala/org/scalameter/Context.scala]: Please avoid overloading - use pattern matching within `apply` instead.

Comment (kjanosz) [inline scalameter-core/src/main/scala/org/scalameter/Context.scala]: Well, I guess that this means that I should either uses intersection type or change `apply` signature to:
```
def apply[T](key: Key[T])
```
and throw error on runtime if `key` is not `KeyWithDefault` or `KeyWithDefaultKey`
What solution do you prefer?

Comment (axel22) [inline scalameter-core/src/main/scala/org/scalameter/Context.scala]: You could use intersection types, or you could introduce their common superclass, something like `UbiquitousKey` -- I'm taking suggestions for the name.

**Comment by kjanosz** [comment]
@axel22: Fixed.

**Comment by axel22** [comment]
LGTM, nice work!

**Pull request merged**